### PR TITLE
Feature/zookeeper backend

### PR DESCRIPTION
**Pull request by calston**

### CODE REVIEW

File: rhumba/backends/zk.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+                qlen = yield self.queueSize(qname)
+
+                stats = yield self.getQueueMessageStats(qname)
+
+                d['queues'][qname] = {
+                    'waiting': qlen,
+                    'messages': stats
+                }
+
+        defer.returnValue(d)
```
Comment (jerith) [inline rhumba/backends/zk.py]: Can we break up this gigantic method a bit? Maybe factor out some of the loop bodies into separate functions.

File: setup.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 setup(
     name="rhumba",
-    version='0.1.18',
+    version='0.2.0',
```
Comment (jerith) [inline setup.py]: Non-release revisions probably shouldn't use "production" version numbers. I prefer to have `1.2.3a0` between releases `1.2.2` and `1.2.3` (and update it to `1.3.0a0` if we make changes big enough to move to `1.3.0` on the next release). That also has the nice benefit that the tagged commit for the release is just the version number change.

File: rhumba/backends/zk.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+from txzookeeper.client import ZookeeperClient
+from txzookeeper.queue import Queue, QueueItem
+
+import zookeeper
+
+
+class Backend(RhumbaBackend):
+    """
+    Rhumba redis backend
```
Comment (jerith) [inline rhumba/backends/zk.py]: Isn't this the "Rhumba zookeeper backend"?

**Comment by jerith** [comment]
I haven't done a thorough review of the new zookeeper stuff (and I'm insufficiently familiar with zookeeper to do that properly), but I'm happy with the refactor of the redis backend and using that as a default.

File: setup.py (hunk 2 above)
Comment (calston) [inline setup.py]: No I fucking hate letters in my versions. It causes all kinds of issues with things that fail to parse it

Comment (jerith) [inline setup.py]: Letters in that part of the version are part of the standard used by pip and friends (and they indicate "prerelease" versions that aren't installed without an explicit version number or a `--pre` flag), and my suggestion is specifically to use them for things that are never published to pypi.

(It was also just a suggestion. I'm happy with whatever you decide to do here.)

**Comment by jerith** [comment]
:+1:

**Pull request merged**